webapp/channels/src/components/widgets/loading/loading_spinner.tsx

```diff
@@ -12,26 +12,22 @@ type Props = {
     style?: React.CSSProperties;
 }
 
-export default class LoadingSpinner extends React.PureComponent<Props> {
-    public static defaultProps: Props = {
-        text: null,
-    };
+const LoadingSpinner: React.FC<Props> = ({ text = null, style }) => {
+    return (
+        <span
+            id='loadingSpinner'
+            className={'LoadingSpinner' + (text ? ' with-text' : '')}
+            style={style}
+            data-testid='loadingSpinner'
+        >
+            <LocalizedIcon
+                className='fa fa-spinner fa-fw fa-pulse spinner'
+                component='span'
+                title={{ id: t('generic_icons.loading'), defaultMessage: 'Loading Icon' }}
```

Comment:
We want to get rid of the `t` function, so do you mind moving this outside of the component and define it with defineMessage?

Something like...

```
const IconTitle = defineMessage({id: 'generic_icons.loading', defaultMessage: 'Loading Icon'});
const LoadingSpinner: React.FC<Props> = ({ text = null, style }) => {
    ...
        title={IconTitle}
```
The other good thing about this is that you only define the string once, so you always pass the same object to the icon, avoiding potential re-renders down the line, and you put less strain in the GC, having to clean up the old message after every re-render.